### DISCUSSION TITLE
User Stories to explore as PoC Workflows

### DISCUSSION THREAD

**Discussion by foundrytom** [OpenAssetIO/OpenAssetIO#611] [Ideas and change proposals]
A gathering point to identify candidate workflows to use as PoCs around OpenAssetIO integrations.

**Comment by antirotor** [comment]
From OpenPype point of view first step would be to implement manager per host (dcc) resolving "representations" for loader plugins dealing with specific file types and data types.

User selects in dialog what he wants to load and what loader should be used, that loader has already prepared context from which it gets the specific file. As a first step, this process can be done using OpenAssetIO to resolve the path for the loader context.

That is the most simple (and not that useful) step. Second will be to integrate OpenAssetIO to publishing process, specifically to Integrator plugins that take care of handling files during the publishing, pushing them to their final location and renaming as necessary.

These two steps will allow to test various types of data passing through and will help to better identify and specify needed Traits/Specifications.

**Reply by foundrytom** [reply]
>  to implement manager per host (dcc)

What would you see as changing per-host? The `context.locale` can be used to adapt behaviour, but it'd be great to get out of having to code per-DCC logic into the manager plugins through agreed traits. Leaving the tool-specific logic to the loader plugins.

**Reply by antirotor** [reply]
yep, what I've meant is test it in one host as a first step and then abstract it to others.

**Reply by foundrytom** [reply]
Ahh perfect, sorry for the misunderstanding!

**Comment by foundrytom** [comment]
We sketched out some ideas for OTIO [a little while back](https://github.com/OpenAssetIO/OpenAssetIO-WG/tree/main/users/2022-08-31):

 - Resolve `target_url` via OpenAssetIO for basic `ExternalReference` resolution.
 - Query related references for proxies to build multi-ref OTIO files from `Clip` metadata.

**Reply by foundrytom** [reply]
Finally had some time to pick this up, now we have all the prerequisite tooling in place to make these sketches pretty trivial from python.  https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/21 has an sketch of using `getRelatedReferences` to retrieve proxy/original versions of some entity, so you could then create a multi-ref OTIO file.
